### PR TITLE
deployment: carry over the securityContext from deployer to lifecycle hooks

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/12103

### CODE REVIEW

**Comment by mfojtik** [comment]
@pweil- @kargakis FYI

**Comment by kargakis** [comment]
There was some discussion upstream related to securityContext for contaienrs being broken but I can't remember if it was an issue or from the mailing list.

**Comment by mfojtik** [comment]
@kargakis if it was broken the deployer pod won't be working?

**Comment by kargakis** [comment]
Not necessarily. Found the discussion: https://github.com/kubernetes/kubeadm/issues/107

**Comment by mfojtik** [comment]
@pweil- @php-coder can you confirm the securityContext is broken for SELinux?

@kargakis also it is possible to set `securityContext` for the deployer pod? If not we are just copying nil.

**Comment by pweil-** [comment]
> can you confirm the securityContext is broken for SELinux?

looks like this was new to 1.12.5/1.13

File: pkg/deploy/strategy/support/lifecycle.go [hunk 1]
```diff
@@ -397,6 +397,7 @@ func makeHookPod(hook *deployapi.LifecycleHook, rc *kapi.ReplicationController,
 					Env:             mergedEnv,
 					Resources:       resources,
 					VolumeMounts:    volumeMounts,
+					SecurityContext: baseContainer.SecurityContext,
```
Comment (pweil-) [inline pkg/deploy/strategy/support/lifecycle.go]: this is a pointer, should we be making a copy?

Comment (mfojtik) [inline pkg/deploy/strategy/support/lifecycle.go]: yes and should not be copied from baseContainer but from the deployment config...

**Comment by mfojtik** [comment]
[test]

@pweil- updated, ptal

File: pkg/deploy/strategy/support/lifecycle.go [hunk 2]
```diff
@@ -375,6 +375,11 @@ func makeHookPod(hook *deployapi.LifecycleHook, rc *kapi.ReplicationController,
 
 	gracePeriod := int64(10)
 
+	podSecurityContext, err := kapi.Scheme.DeepCopy(rc.Spec.Template.Spec.SecurityContext)
```
Comment (pweil-) [inline pkg/deploy/strategy/support/lifecycle.go]: @mfojtik @kargakis so the pod SC and the container SC can differ.  The lifecycle hook targets a container to get the image.  Do you think it is reasonable to use the container SC as well?  Ie. set both the pod SC and container SC or just the pod SC.

Comment (mfojtik) [inline pkg/deploy/strategy/support/lifecycle.go]: @pweil- if DC has RC with more than 1 container, which one will you pick for the hook pod? ;-)

Comment (pweil-) [inline pkg/deploy/strategy/support/lifecycle.go]: you have to tell the hook pod the container name so it pulls the image.  That would be unique

Comment (mfojtik) [inline pkg/deploy/strategy/support/lifecycle.go]: fixed

File: pkg/deploy/strategy/support/lifecycle.go (hunk 1 above)
Comment (mfojtik) [inline pkg/deploy/strategy/support/lifecycle.go]: @pweil- fixed. finally ;)

**Comment by mfojtik** [comment]
@pweil- updated, PTAL :)

**Comment by php-coder** [comment]
I'd also update the comment here: https://github.com/openshift/origin/blob/83e325079412e5c64c1effa2040d58312a4ddf7d/pkg/deploy/strategy/support/lifecycle.go#L176-L181

**Comment by mfojtik** [comment]
good catch.

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**